s3/pkg/service/service.go

```diff
@@ -302,6 +302,14 @@ func (s *s3Service) GetTierMap(ctx context.Context, in *pb.BaseRequest, out *pb.
 func (s *s3Service) UpdateBucket(ctx context.Context, in *pb.Bucket, out *pb.BaseResponse) error {
 	log.Info("UpdateBucket is called in s3 service.")
 
+	//update versioning if not nil
+	if in.Versioning != nil {
+		err := s.MetaStorage.Db.UpdateBucketVersioning(ctx, in.Name, in.Versioning.Status)
```

Comment:
What if user create a bucket without versioning, and then put versioning to this bucket? In this scenario, nothing will happen by update, because you created a new table to save versioning and no record in this table.

Reply:
When user will get the interface of createbucket on dashboard there will be a button to select versioning enabled or disabled by default, so always we will get one status value, hence it is handled like that

Reply:
Then in the scenario I mentioned above, which API is supposed to be used? No record in the new table you created while updating it, so nothing will happen.

Reply:
Sure will check

Reply:
Hi @nguptaopensds how about this?